pkg/licensing/normalize_test.go

```diff
@@ -1,13 +1,47 @@
 package licensing_test
 
 import (
+	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 
 	"github.com/aquasecurity/trivy/pkg/licensing"
 )
 
+func TestNormalize(t *testing.T) {
+	tests := []struct {
+		licenses   []string
+		normalized string
+	}{
+		{
+			[]string{
+				"APACHE",
+				" APACHE ",
+				"APACHE License",
+				"The Apache License",
+				"THE APACHE LICENSE",
+				"  THE APACHE LICENSE  ",
+			},
+			"Apache-1.0",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.normalized, func(t *testing.T) {
+			for _, ll := range tt.licenses {
+				res := licensing.Normalize(ll)
+				assert.Equal(t, tt.normalized, res)
+			}
+		})
+	}
+	t.Run("All mapping keys must be uppercase", func(t *testing.T) {
+		for k := range licensing.Mapping {
+			res := strings.ToUpper(k)
+			assert.Equal(t, k, res)
+		}
+	})
+}
```

Comment:
I think we can omit this test.
I think we'll just ask contributors to add new licenses to the `tests`.

Can you remove this test and revert `Mapping` to `mapping`

Reply:
`mapping `is back to private.

I also removed additional "+", "-only" and "-or-later" suffixes from https://spdx.dev/learn/handling-license-info/

Reply:
> mapping is back to private.

Thanks!

> I also removed additional "+", "-only" and "-or-later" suffixes from https://spdx.dev/learn/handling-license-info/

SPDX differentiates between `GPL-1.0-only` and `GPL-1.0-or-later` (see https://spdx.org/licenses/).
So I'm not sure we need to do this.

Reply:
1. There is no change in behavior, for example `GPL-2.0-ONLY` and `GPL-2.0-OR-LATER` are already mapped to GPL20 in current trivy mapping
2. `GPL-1.0-only` and `GPL-1.0-or-later` are not in current [category.go](https://github.com/aquasecurity/trivy/blob/4dc507028ea75a8e1ed319b4df5d865e9e5d2b54/pkg/licensing/category.go), and it is probably not necessary to add them because of next point:
3. The main purpose of the normalization is to classify licenses into categories in [category.go](https://github.com/aquasecurity/trivy/blob/4dc507028ea75a8e1ed319b4df5d865e9e5d2b54/pkg/licensing/category.go). `GPL-1.0-only`, `GPL-1.0-or-later` and `GPL-1.0` would have the same category.

Reply:
hmm... you're right.
Thank you for the clarification!